es-ds-components/components/es-toggle.vue

```diff
@@ -0,0 +1,158 @@
+<script setup lang="ts">
+import { SwitchRoot, SwitchThumb } from 'reka-ui';
+
+const model = defineModel<boolean>({
+    default: false,
+});
+
+defineProps({
+    disabled: {
+        type: Boolean,
+        default: false,
+    },
+    ariaLabel: {
+        type: String,
+        default: 'es-toggle',
+        required: false,
+    },
+    label: {
+        type: String,
+        default: '',
+        required: false,
+    },
+});
+
+// Workaround: Reka UI components don't play nice with the Nuxt 3.16+ dev server side rendering while using NPM link
+const wrappingComponent = import.meta.dev ? resolveComponent('ClientOnly') : 'span';
+</script>
+
+<template>
+    <component :is="wrappingComponent">
+        <div
+            class="d-flex"
+            :class="{ 'has-text': $slots.text }">
+            <div
+                class="d-flex flex-column align-items-center"
+                :class="{ 'has-value': label }">
+                <switch-root
+                    v-model="model"
+                    :disabled="disabled"
+                    :aria-label="ariaLabel"
+                    :aria-labelledby="label ? 'es-toggle-value' : undefined"
+                    class="es-toggle">
+                    <switch-thumb class="es-toggle-thumb" />
+                </switch-root>
+                <p
+                    v-if="label"
+                    id="es-toggle-value"
+                    class="es-toggle-value">
+                    {{ label }}
+                </p>
+            </div>
+            <div
+                v-if="$slots.text"
+                :class="{ 'cursor-pointer': !disabled }"
+                @click="!disabled && (model = !model)">
+                <slot name="text" />
+            </div>
```

Comment:
I think we may actually want this to look something like this, given `es-toggle-id` is the `id` attribute on `<switch-root>`. I pulled this example from the code in the [first example from the Reka UI docs](https://reka-ui.com/docs/components/switch#switch). This will provide the same functionality as the `@click` event, and more (e.g. hovering, focusing, etc).
```suggestion
            <label
                v-if="$slots.text"
                for="es-toggle-id">
                <slot name="text" />
            </label>
```
This will require each instance of `es-toggle` to have a unique `id` attribute that gets defined or passed in, which I think Nathaniel called out in an earlier review!